snuba/web/db_query.py

```diff
@@ -530,6 +567,19 @@ def record_cache_hit_type(hit_type: int) -> None:
     )
 
 
+def write_query_cache_hit_metric(
+    is_sorted: str, reader: Reader, key_sorted: str
+) -> None:
+    metrics.increment(
+        "snuba.api.query.cache_hit",
+        tags={
+            "partition_id": reader.cache_partition_id or "default",
+            "query_info": is_sorted,
+            "key": key_sorted,
```

Comment:
You cannot use `query_id`  as a key here. Datadog only allows ~300 unique tag values in a metric in total. Please remove this.